src/util/thread-helper.ts

```diff
@@ -0,0 +1,35 @@
+import { Worker, WorkerOptions } from 'worker_threads';
+import path from 'path';
+
+export function runThreaded(scriptPath: string, options: WorkerOptions) {
+  const resolvedPath = require.resolve(scriptPath);
+
+  const isTs = /\.ts$/.test(resolvedPath);
+
+  const ex = isTs
+    ? `
+    const wk = require('worker_threads');
+    require('tsconfig-paths/register');
+    require('ts-node').register({
```

Comment:
Would this impact any environments that `ts-node` is now a runtime dependency? And does this work across node runtimes, including `bun`?

Reply:
this is wonky. the inline execution is only for development. the published sdk will resolve a js file which is called directly

added a note